plugins/inputs/sqlserver/sqlserver.go

```diff
@@ -433,40 +484,36 @@ BEGIN
 		,COALESCE(mf.[physical_name],''RBPEX'') AS [physical_filename]	--RPBEX = Resilient Buffer Pool Extension
 		,COALESCE(mf.[name],''RBPEX'') AS [logical_filename]	--RPBEX = Resilient Buffer Pool Extension	
 		,mf.[type_desc] AS [file_type]
-		,IIF( RIGHT(vs.[volume_mount_point],1) = ''\''	/*Tag value cannot end with \ */
-			,LEFT(vs.[volume_mount_point],LEN(vs.[volume_mount_point])-1)
-			,vs.[volume_mount_point]
-		) AS [volume_mount_point]
 		,vfs.[io_stall_read_ms] AS [read_latency_ms]
 		,vfs.[num_of_reads] AS [reads]
 		,vfs.[num_of_bytes_read] AS [read_bytes]
 		,vfs.[io_stall_write_ms] AS [write_latency_ms]
 		,vfs.[num_of_writes] AS [writes]
-		,vfs.[num_of_bytes_written] AS [write_bytes]
-		'
-		+ 
-		CASE
-			WHEN LEFT(CAST(SERVERPROPERTY('ProductVersion') AS nvarchar) ,2) = '11'
-				/*SQL Server 2012 (ver 11.x) does not have [io_stall_queued_read_ms] and [io_stall_queued_write_ms]*/
-				THEN ''
-				ELSE N',vfs.io_stall_queued_read_ms AS [rg_read_stall_ms] ,vfs.io_stall_queued_write_ms AS [rg_write_stall_ms]'
-		END 
-		+
-	N'FROM sys.dm_io_virtual_file_stats(NULL, NULL) AS vfs
+		,vfs.[num_of_bytes_written] AS [write_bytes]'
+		+ @Columns + N'
+	FROM sys.dm_io_virtual_file_stats(NULL, NULL) AS vfs
 	INNER JOIN sys.master_files AS mf WITH (NOLOCK)
 		ON vfs.[database_id] = mf.[database_id] AND vfs.[file_id] = mf.[file_id]
-	CROSS APPLY sys.dm_os_volume_stats(vfs.[database_id], vfs.[file_id]) AS vs
 	'
+	+ @Tables;
+	
+	/*Debug only*/
+	--PRINT @SqlStatement
+	
 	EXEC sp_executesql @SqlStatement
 
 END
-
 `
 
 // Conditional check based on Azure SQL DB, Azure SQL Managed instance OR On-prem SQL Server
 // EngineEdition=5 is Azure SQL DB, EngineEdition=8 is Managed instance
 
```

Comment:
This complexity isn't worth it as we have taken a query that you could copy/paste and run to one that has dynamic SQL and to see the SQL generated you have to print it.  If there was already dynamic SQL I get changing it to tidy up but if not I wouldn't make a change to dynamic SQL unless is providing value.